client/src/components/MidiPlayer.jsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from "react";
+import { blobToNoteSequence } from "@magenta/music";
+
+const MidiPlayer = (midiFile) => {
+  useEffect(() => {
+    if (!midiFile.midiFile) {
+      return;
+    }
+    blobToNoteSequence(midiFile.midiFile).then((newNoteSeq) => {
+      const midiPlayer = document.getElementById("midiPlayer");
+      midiPlayer.noteSequence = newNoteSeq;
+    });
+  }, [midiFile]);
```

Comment:
this whole thing is a little sloppy, but it works so whatever.

obviously using document.getElementById is stupid, but this midi-player is a custom element and doesn't forward it's reference, so I don't think you can just do
```
const ref = useRef();

      <midi-player
        ref={ref}
        id="midiPlayer"
        noteSequence={null}
        visualizer="#myVisualizer"
      ></midi-player>
```

I also didn't know how to import blobToNoteSequence from an npm install, so that's one of the reasons I didn't want to install them with npm. if someone knows how lay it on me.